keps/sig-node/2621-cpu-allocation-llc-affinity/README.md

```diff
@@ -105,28 +105,32 @@ We make a bench with `stream2` DAXPY, as we can see, cross ccx(cross uncore-cach
 ![stream2_daxpy](stream2_daxpy.png "stream2_daxpy")
 
 And, when workload is memory sensitive, this feature can improve memory bandwidth significantly(20% above).
-
+We also conduct tests in real applications like dpdk and redis, and get performance data. While in dpdk, with throughput degration from 19Mpps(share uncore-cache) to 12Mpps(cross uncore-cache).
 ### Notes/Constraints/Caveats (Optional)
 
 ### Risks and Mitigations
 
 + Currently no risks was found.
-+ Feature is enbled by a gate - a new kube feature with default false, potential risk effects could be limited.
++ Feature is enabled by a gate - a new kube feature with default false, potential risk effects could be limited.
 
 ## Design Details
 
 - Feature Gate
-  - Add `CPUManagerUncoreCacheAlign` to kubelet's feature-gates to enable(true)/disable(false) the feature.
-  - Also, more than one l3 cache should exist in a single socket/package.
-
+  - C1: Add `CPUManagerUncoreCacheAlign` to kubelet's feature-gates to enable(true)/disable(false) the feature.
```

Comment:
So this feature (per my previous comment, see history) will need to depend on a feature gate. But you don't necessarily need a separate feature gate, you can depend on the one we added on https://github.com/kubernetes/enhancements/pull/2626.

This CPUManager feature gate will most likely be alpha in the 1.23 cycle, which fit the needs of this KEP.

I think you can conditionally enable this optimization depending on a cpu manager policy options. This way you keep the conditional logic you already need to support the feature gate, without extra burden.

Fitting in the new CPUManager policy options is also a very nice and clean design.

Reply:
hi, @fromanirh  since i've post a patch <https://github.com/kubernetes/kubernetes/pull/102307>, it maybe help you to understand why i choose a kubelet running option other than a separate cpu manager policy. IMHO, attach to a policy is also optional, but it will cause some redundant logic.

Reply:
Thanks for sharing the implementation. I don't see yet where we need redundant logic, however. We could:

1. get the enable/disable flag from cpumanager options. Probably the option should be on by default
2. propagate the flag down to the `cpuAccumulator`
3. consume the flag into `isUncoreCacheAlignEnabled`

IMHO this is also a nicer and more integrated design.
Now: implementation wise, this flow seems compliant with the production-readiness review (see the details in https://kubernetes.slack.com/archives/CPNHUMN74/p1620312071045800) because new features should depend on _A_ compatible and related feature gate; a new feature gate would be alpha level, and the `cpuManagerOptions` feature gate will still be alpha in 1.23.

So reshaping to use the cpuManagerOptions still seems a totally valid option and I think is cleaner from overall design perspective.
Let's see what other reviewers (@klueska :) ) think, and please let me know if there are concerns or requirements that I missed.